mobile/test/common/integration/sds_integration_test.cc

```diff
@@ -16,77 +14,80 @@ namespace {
-      setUpSdsConfig(secret_config, "client_cert");
-
-      transport_socket->set_name("envoy.transport_sockets.tls");
-      transport_socket->mutable_typed_config()->PackFrom(tls_context);
-    });
+    ExtensionRegistry::registerFactories();
+
+    XdsIntegrationTest::initialize();
+    initializeXdsStream();
   }
 
-  void createUpstreams() override {
-    // This is the fake upstream configured with SSL for the base cluster.
-    addFakeUpstream(Ssl::createUpstreamSslContext(context_manager_, *api_), upstreamProtocol(),
-                    /*autonomous_upstream=*/true);
+  void createEnvoy() override {
+    const std::string target_uri = Network::Test::getLoopbackAddressUrlString(ipVersion());
+    Platform::XdsBuilder xds_builder(target_uri,
+                                     fake_upstreams_.back()->localAddress()->ip()->port());
+    xds_builder.addClusterDiscoveryService();
+    builder_.setXds(std::move(xds_builder));
+    XdsIntegrationTest::createEnvoy();
   }
 
+  void createUpstreams() override { addFakeUpstream(Http::CodecType::HTTP2); }
+
 protected:
+  void sendCdsResponse() {
+    auto cds_cluster = createSingleEndpointClusterConfig(std::string(XDS_CLUSTER_NAME));
+    // Update the cluster to use SSL.
+    auto* transport_socket = cds_cluster.mutable_transport_socket();
+    envoy::extensions::transport_sockets::tls::v3::UpstreamTlsContext tls_context;
+    tls_context.set_sni("lyft.com");
+    auto* secret_config =
+        tls_context.mutable_common_tls_context()->add_tls_certificate_sds_secret_configs();
+    setUpSdsConfig(secret_config, SECRET_NAME);
+    transport_socket->set_name("envoy.transport_sockets.tls");
+    transport_socket->mutable_typed_config()->PackFrom(tls_context);
+    sendDiscoveryResponse<envoy::config::cluster::v3::Cluster>(
+        Config::TypeUrl::get().Cluster, {cds_cluster}, {cds_cluster}, {}, "55");
+  }
+
+  void initializeXdsStream() override {
+    AssertionResult result = fake_upstreams_.back()->waitForHttpConnection(
+        *BaseIntegrationTest::dispatcher_, BaseIntegrationTest::xds_connection_);
+    RELEASE_ASSERT(result, result.message());
+
+    result = BaseIntegrationTest::xds_connection_->waitForNewStream(
+        *BaseIntegrationTest::dispatcher_, xds_stream_);
+    RELEASE_ASSERT(result, result.message());
+    xds_stream_->startGrpcStream();
+  }
+
   void sendSdsResponse(const envoy::extensions::transport_sockets::tls::v3::Secret& secret) {
     envoy::service::discovery::v3::DiscoveryResponse discovery_response;
     discovery_response.set_version_info("1");
     discovery_response.set_type_url(Config::TypeUrl::get().Secret);
     discovery_response.add_resources()->PackFrom(secret);
-
     xds_stream_->sendGrpcMessage(discovery_response);
   }
 
   void setUpSdsConfig(envoy::extensions::transport_sockets::tls::v3::SdsSecretConfig* secret_config,
-                      const std::string& secret_name) {
+                      const absl::string_view secret_name) {
```

Comment:
what's the motivation to declare string_view as const here? (i don't think it's idiomatic to declare string_view parameters with const, since they are just read-only containers for the underlying string anyway)

Reply:
Oops. Didn't mean to add `const` :)